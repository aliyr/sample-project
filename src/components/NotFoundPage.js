import React , {Component} from 'react';
import {Link} from 'react-router-dom';

class NotFoundPage extends Component{
    render() {

        return (
            <div>
                <h1>
                    404!<Link to="/">go home</Link>
                </h1>
            </div>
        )
    }
}

export default NotFoundPage;

