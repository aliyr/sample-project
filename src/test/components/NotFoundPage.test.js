import React from 'react';
import NotFoundPage from '../../components/NotFoundPage';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import {shallow} from 'enzyme';

test('test not found page',() => {
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<NotFoundPage/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    const wrapper = shallow(<NotFoundPage/>);
    expect(wrapper).toMatchSnapshot();

});
