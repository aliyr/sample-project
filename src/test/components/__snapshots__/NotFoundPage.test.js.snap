// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test not found page 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        404!
        <Link
            replace={false}
            to="/"
        >
            go home
        </Link>
    </h1>
</div>,
  "nodes": Array [
    <div>
      <h1>
            404!
            <Link
                  replace={false}
                  to="/"
            >
                  go home
            </Link>
      </h1>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <NotFoundPage />,
    "_forcedUpdate": false,
    "_instance": NotFoundPage {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <h1>
            404!
            <Link
                  replace={false}
                  to="/"
            >
                  go home
            </Link>
      </h1>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NotFoundPage />,
}
`;
